rfc/0014-technical-teams.md

```diff
@@ -0,0 +1,79 @@
+# Technical Team Structure
+
+## Summary
+
+This RFC describes the structure and processes that technical teams of the
+Quilt project will follow. It does not detail members of each team nor does
+it restrict them from committing to additional policies in the future.
+
+## Motivation
+
+Laying out a clean, formal specification for how people will interact with
+and decide on changes will help technical processes run more smoothly, from
+creating entirely new projects to making small revisions to existing ones.
+
+## Explanation
+
+### Technical Leads
+Technical leads are the highest technical authority in the Quilt Project.
+They are not aligned to any particular technical team, but instead act as
+guides, steering the other teams in a direction that's beneficial to Quilt
+as a whole. While the administrative board exists to oversee and facilitate
+policy changes throughout the project, technical leads exist to oversee and
+facilitate larger technical decisions.
+
+### Team Responsibilities
+Technical teams are typically aligned to a specific project or set of
+projects. They are responsible for reviewing pull requests and ensuring that
+they meet the Quilt projects goals in quality and purpose. Once a team has 
+decided that a pull request is suitable for merging, a technical lead will be 
+responsible for the actual merging.
+
+### Technical Teams
+The following is a list of top-level technical teams.
+
+| Team | Repositories | Notes
+|-|-|-|
+| Quilt Loader | [Quilt Loader](https://github.com/QuiltMC/quilt-loader)<br>[Sat4j](https://github.com/QuiltMC/quilt-loader-sat4j)<br>[Access Widener](https://github.com/QuiltMC/access-widener)<br>[Mixin](https://github.com/QuiltMC/Mixin)<br>[Sponge Mixin Compile Extensions](https://github.com/QuiltMC/sponge-mixin-compile-extensions)<br>[Quilt JSON5](https://github.com/QuiltMC/quilt-json5)
```

Comment:
Should Mixin be under the responsibility of ASMR (or maybe give both teams "control")? We are expecting to be doing some active maintenance on it to give access to a couple ASMR features early, and we will also be responsible for the Mixin Compilers

Reply:
Since (for now) Mixin has nothing to do with ASMR, I think it being in loader is fine (for now).

Once the ASMR-Mixin variant starts to exist, it will probably still be separate from Sponge-Mixin until it completely replaces it.

Same goes for AW.

Reply:
It will make sense to have all of the bytecode manipulation libraries under the same umbrella eventually. For now, while work on mixin and AW is purely maintenance and no actual work is being done on them, it makes sense to me to keep them under the Loader umbrella.